File: API/Features/Reservations/Resources/Form/ReservationReadResource.cs
using System.Collections.Generic;
using API.Infrastructure.Classes;

namespace API.Features.Reservations {

    public class ReservationReadResource {

        public string ReservationId { get; set; }

        public string Date { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public int TotalPersons { get; set; }
        public string TicketNo { get; set; }
        public string Email { get; set; }
        public string Phones { get; set; }
        public string Remarks { get; set; }
        public string UserId { get; set; }

        public SimpleResource Customer { get; set; }
        public SimpleResource Destination { get; set; }
        public PickupPointWithPortDropdownResource PickupPoint { get; set; }
        public SimpleResource Driver { get; set; }
        public SimpleResource Ship { get; set; }

        public List<PassengerReadResource> Passengers { get; set; }

    }

}

File: FrontEnd/src/app/features/reservations/classes/resources/form/reservation/reservation-read-resource.ts
import { Guid } from 'guid-typescript'
// Custom
import { CustomerDropdownDTO } from '../../../../../customers/classes/dtos/customer-dropdown-dto'
import { DestinationDropdownDTO } from '../../../../../destinations/classes/dtos/destination-dropdown-dto'
import { DriverDropdownDTO } from '../../../../../drivers/classes/dtos/driver-dropdown-dto'
import { PassengerReadResource } from '../passenger/passenger-read-resource'
import { PickupPointDropdownDTO } from '../../../../../pickupPoints/classes/dtos/pickupPoint-dropdown-dto'
import { ShipDropdownResource } from '../dropdown/ship-dropdown-resource'
import { PortDropdownDTO } from 'src/app/features/ports/classes/dtos/port-dropdown-dto'

export class ReservationReadResource {

    reservationId: Guid

    date: string
    refNo: string
    email: string
    phones: string
    remarks: string
    adults: number
    kids: number
    free: number
    totalPersons: number
    ticketNo: string

    customer: CustomerDropdownDTO
    destination: DestinationDropdownDTO
    pickupPoint: PickupPointDropdownDTO
    driver: DriverDropdownDTO
    ship: ShipDropdownResource
    port: PortDropdownDTO

    passengers: PassengerReadResource

}

